Makefile

```diff
@@ -1,191 +1,517 @@
-# Export SPIRE_DEV_HOST to docker to build SPIRE using a docker container
-ifeq ($(SPIRE_DEV_HOST), docker)
-	docker = docker run -v $(docker_volume_spire) -v $(docker_volume_gopath) -it $(docker_image)
-	container = container
-	dev_host_status = "SET"
+DIR := ${CURDIR}
+
+# There is no reason GOROOT should be set anymore. Unset it so it doesn't mess
+# with our go toolchain detection/usage.
+ifneq ($(GOROOT),)
+	export GOROOT=
+endif
+
+cyan := $(shell which tput > /dev/null && tput setaf 6 2>/dev/null || echo "")
+reset := $(shell which tput > /dev/null && tput sgr0 2>/dev/null || echo "")
+bold  := $(shell which tput > /dev/null && tput bold 2>/dev/null || echo "")
+
+.PHONY: default all help
+
+default: build
+
+all: build lint test
+
+help:
+	@echo "$(bold)Usage:$(reset) make $(cyan)<target>$(reset)"
+	@echo
+	@echo "$(bold)Build:$(reset)"
+	@echo "  $(cyan)build$(reset)                         - build all SPIRE binaries"
+	@echo "  $(cyan)artifact$(reset)                      - build SPIRE tarball artifact"
+	@echo
+	@echo "$(bold)Test:$(reset)"
+	@echo "  $(cyan)test$(reset)                          - run unit tests"
+	@echo "  $(cyan)race-test$(reset)                     - run unit tests with race detection"
+	@echo "  $(cyan)integration$(reset)                   - run integration tests (requires docker images)"
+	@echo
+	@echo "$(bold)Build and test:$(reset)"
+	@echo "  $(cyan)all$(reset)                           - build all SPIRE binaries, lints the code, and runs unit tests"
+	@echo
+	@echo "$(bold)Docker image:$(reset)"
+	@echo "  $(cyan)images$(reset)                        - builds all SPIRE Docker images"
+	@echo "  $(cyan)spire-server-image$(reset)            - builds SPIRE server Docker image"
+	@echo "  $(cyan)spire-agent-image$(reset)             - builds SPIRE agent Docker image"
+	@echo "  $(cyan)k8s-workload-registrar-image$(reset)  - builds Kubernetes Workload Registrar image"
```

Comment:
nit: it would be nice to consistently refer to a "Docker image" and not just "image" throughout this help

Reply:
👍